### ISSUE TITLE
Client `Logger.colored` can't be called with the default responseColor

### ISSUE THREAD

**Issue by kubukoz** [http4s/http4s#4647]
When you try this:

```scala
val aa = Logger.colored[IO](
  logHeaders = true,
  logBody = false,
  logAction = Some(s => IO.println(s))
) _
```

The result is:

```
type mismatch;
 found   : org.http4s.Response[Nothing] => String
 required: org.http4s.Response[cats.effect.IO] => String
Error occurred in an application involving default arguments.
```

Providing the default `responseColor: Response[F] => String = ResponseLogger.defaultResponseColor _` makes it compile:

```scala
val aa = Logger.colored[IO](
  logHeaders = true,
  logBody = false,
  responseColor = ResponseLogger.defaultResponseColor _,
  logAction = Some(s => IO.println(s))
) _
```

Scala 2.13.5, http4s 1.0.0-M19

**Comment by rossabaker** [comment]
Is there a workaround that would let us elide that argument?  I would strongly prefer not to have an overload.

I guess if we can't use the default, we should remove the default.

**Comment by danicheg** [comment]
This was fixed in a6708bd8e0896547616c43f169bc6ab6bf640810
